pkg/shared/utils/sync/distributed_lock_mgr.go

```diff
@@ -0,0 +1,69 @@
+package sync
+
+import (
+	"fmt"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+	"gorm.io/gorm"
+	"sync"
+)
+
+var mgrLock sync.Mutex
+
+var _ DistributedLockMgr = &distributedLockMgr{}
+
+func NewDistributedLockMgr(db *gorm.DB) DistributedLockMgr {
+	return &distributedLockMgr{
+		db:           db,
+		locks:        map[string][]DistributedLock{},
+		createLockFn: NewDistributedLock,
+	}
+}
+
+// DistributedLockMgr manages a list of DistributedLock objects. The DistributedLock object is automatically removed
+// from the list when the Unlock method is called.
+type DistributedLockMgr interface {
```

Comment:
What is the advantage of using this over the DistributedLock type?
That type already blocks when trying to get a specific lockID

Reply:
As discussed previously, locks must be released by the `DistributedLock` instance that created them.

In the `secure storage`, two methods handle `lock` and `unlock` operations for a specific `lockID`. That means that without `DistributedLockMgr`, all the logic for managing the `DistributedLock` list would need to be implemented in the `secure lock`. 

Since this is a common scenario, I think using `DistributedLockMgr` is a better solution.